tiledb/sm/serialization/query.cc

```diff
@@ -1582,6 +1594,16 @@ Status query_to_capnp(
     }
   }
 
+  // The server should throw if it's about to serialize an incomplete query
+  // that has aggregates on it, this behavior is currently not supported.
+  if (!client_side && query.status() == QueryStatus::INCOMPLETE &&
+      !query.get_default_channel_aggregates().empty()) {
+    throw Status_SerializationError(
+        "Aggregates are not currently supported in incomplete remote "
+        "queries");
+  }
+  query_aggregates_to_capnp(query, query_builder);
```

Comment:
This will unconditionally create the aggregate capnproto objects. Typical we want to protect and not initialize capnproto classes if they are not used. Either here or in the function we need to not hard code and assumption of 1 channel and support the cases of >=0 channels.